src/test/java/com/epam/test_generator/services/ProjectServiceTest.java

```diff
@@ -0,0 +1,398 @@
+        projectService.removeProject(777L);
+        verify(projectDAO).delete(SIMPLE_PROJECT_ID);
+    }
+
+    @Test
+    public void addUserToProject_valid_Success() {
+        simpleProject1.setUsers(new HashSet<>());
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        when(userService.getUserById(SIMPLE_USER_ID)).thenReturn(simpleUser1);
+
+        assertFalse(simpleProject1.getUsers().contains(simpleUser1));
+        projectService.addUserToProject(SIMPLE_PROJECT_ID, SIMPLE_USER_ID);
+        assertTrue(simpleProject1.getUsers().contains(simpleUser1));
+        verify(projectDAO).save(simpleProject1);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void addUserToProject_invalidProjectId_NotFound() {
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        when(userService.getUserById(SIMPLE_USER_ID)).thenReturn(simpleUser1);
+
+        projectService.addUserToProject(777L, SIMPLE_USER_ID);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void addUserToProject_invalidUserId_NotFound() {
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        when(userService.getUserById(SIMPLE_USER_ID)).thenReturn(simpleUser1);
+
+        projectService.addUserToProject(SIMPLE_PROJECT_ID, 666L);
+    }
+
+    @Test
+    public void removeUserFromProject_valid_Success() {
+        simpleProject1.setUsers(Sets.newHashSet(simpleUser1));
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        when(userService.getUserById(SIMPLE_USER_ID)).thenReturn(simpleUser1);
+
+        assertTrue(simpleProject1.getUsers().contains(simpleUser1));
+        projectService.removeUserFromProject(SIMPLE_PROJECT_ID, SIMPLE_USER_ID);
+        assertFalse(simpleProject1.getUsers().contains(simpleUser1));
+        verify(projectDAO).save(simpleProject1);
+    }
+
+    @Test (expected = NotFoundException.class)
+    public void removeUserFromProject_invalidProjectId_NotFound() {
+        simpleProject1.setUsers(Sets.newHashSet(simpleUser1));
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        when(userService.getUserById(SIMPLE_USER_ID)).thenReturn(simpleUser1);
+
+        projectService.removeUserFromProject(777L, SIMPLE_USER_ID);
+    }
+
+    @Test (expected = NotFoundException.class)
+    public void removeUserFromProject_invalidUserId_NotFound() {
+        simpleProject1.setUsers(Sets.newHashSet(simpleUser1));
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        when(userService.getUserById(SIMPLE_USER_ID)).thenReturn(simpleUser1);
+
+        projectService.removeUserFromProject(SIMPLE_PROJECT_ID, 666L);
+    }
+
+    @Test
+    public void closeProject_validProjectId_Success() {
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        Project project = simpleProject1;
```

Comment:
reassigning object makes test case hard to figure out, remove reassignment

Reply:
removed